src/op_mode/image_installer.py

```diff
@@ -494,10 +494,10 @@ def get_cli_kernel_options(config_file: str) -> list:
     config = ConfigTree(read_file(config_file))
     config_dict = loads(config.to_json())
     kernel_options = dict_search('system.option.kernel', config_dict)
-    k_cpu_opts = kernel_options.get('cpu', {})
-    k_memory_opts = kernel_options.get('memory', {})
     if kernel_options is None:
         kernel_options = {}
+    k_cpu_opts = kernel_options.get('cpu', {})
+    k_memory_opts = kernel_options.get('memory', {})
     cmdline_options = []
 
```

Comment:
The entire code section could be simplified as `.get()` it not `None` safe.
There is no need to complete the entire code path if there is no `kernel_options` at all and the variable thus equals `None` - we can simply return the empty `cmdline_options` list.

```suggestion
    cmdline_options = []
    kernel_options = dict_search('system.option.kernel', config_dict)
    # bail out early if there are no Kernel CLI options defined
    if kernel_options is None:
        return cmdline_options
```

Reply:
It makes sense in my opinion :+1:

Reply:
@c-po Please make a separate PR for this. I think we need to get a known-working form of the fix merged as soon as possible since the bug has serious impact.